sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/base_node.py

```diff
@@ -231,7 +231,7 @@ def _parse_io(cls, io_dict: dict, parse_cls):
                 value = value._deepcopy()  # Decoupled input and output
                 io_dict[key] = value
                 value.mode = None
-            elif isinstance(value, dict):
+            elif type(value) == dict:  # Use type comparison instead of is_instance to skip _GroupAttrDict
```

Comment:
this looks a little wired for me. Do you know when will the value be a dict?

Reply:
If we're not supposed to support dict, we can just remove this else logic

Reply:
parse_io looks kind of duplicate with self._build_input_attrs, but that is another problem

Reply:
that's a good question, I think I could explain the valid scenario based on my understanding, when we are loading job from yaml files, the inputs and outputs item will be a dict like `{'job_data_path': '${{parent.inputs.pipeline_job_data_path}}'}` , I'll add some comments here.

Reply:
and the `_build_inputs_dict`(I'm not sure if `_build_input_attrs` means this), the function will zip definitions with inputs, it will be called by both nodes and jobs, while `_parse_io` only appeared in BaseNode now, not sure we could merge them.

Reply:
If it's for loading from yaml, shall we move it to InputSchema?

Reply:
Offline synced, will refine it somehow in the next pr